tests/Avalonia.Benchmarks/Styling/ResourceBenchmarks.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using Avalonia.Controls;
+using Avalonia.Platform;
+using Avalonia.PlatformSupport;
+using Avalonia.Styling;
+using Avalonia.Themes.Fluent;
+using Avalonia.UnitTests;
+using BenchmarkDotNet.Attributes;
+using Moq;
+
+namespace Avalonia.Benchmarks.Styling
+{
+    [MemoryDiagnoser]
+    public class ResourceBenchmarks : IDisposable
+    {
+        private readonly Control _searchStart;
+        private readonly IDisposable _app;
+    
+        private static IDisposable CreateApp()
+        {
+            var services = new TestServices(
+                assetLoader: new AssetLoader(),
+                globalClock: new MockGlobalClock(),
+                platform: new AppBuilder().RuntimePlatform,
+                renderInterface: new MockPlatformRenderInterface(),
+                standardCursorFactory: Mock.Of<ICursorFactory>(),
+                styler: new Styler(),
+                theme: () => LoadTheme(),
+                threadingInterface: new NullThreadingPlatform(),
+                fontManagerImpl: new MockFontManagerImpl(),
+                textShaperImpl: new MockTextShaperImpl(),
+                windowingPlatform: new MockWindowingPlatform());
+
+            return UnitTestApplication.Start(services);
+        }
+    
+        private static Styles LoadTheme()
+        {
+            AssetLoader.RegisterResUriParsers();
+            
+            var preHost = new Style();
+            preHost.Resources.Add("preTheme", null);
+            
+            var postHost = new Style();
+            postHost.Resources.Add("postTheme", null);
+            
+            return new Styles
+            {
+                preHost,
+                new FluentTheme(new Uri("avares://Avalonia.Benchmarks")),
```

Comment:
My only worry about this is that changes to the fluent theme could affect benchmarks, which is fine for benchmarking the fluent theme, but this benchmark should really be about resource lookups I think?

Reply:
We can create a placeholder theme. I was basically looking for a large set of styles/resources.

Reply:
Changed to generate styles instead. Got similar benchmark results.

|                  Method |        Mean |     Error |    StdDev | Allocated |
|------------------------ |------------:|----------:|----------:|----------:|
|         FindPreResource | 1,086.21 us | 22.644 us | 66.412 us |       1 B |
|        FindPostResource |    41.99 us |  0.838 us |  2.056 us |         - |
| FindNotExistingResource | 1,129.00 us | 22.471 us | 62.268 us |       1 B |